File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    p1 = [1, 2, 3, 4, 5]
    p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    s1 = s2 = s3 = 0
    for i in range(len(answers)):
        if answers[i] == p1[i%len(p1)]:
            s1 += 1
        if answers[i] == p2[i%len(p2)]:
            s2 += 1
        if answers[i] == p3[i%len(p3)]:
            s3 += 1
    score = [s1, s2, s3]
    for i in range(3):
        if score[i] == max(score):
            answer.append(i+1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] a = {1, 2, 3, 4, 5};
        int[] b = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] c = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int sa = 0;
        int sb = 0;
        int sc = 0;
        for (int i = 0; i < answers.length; i++) {
            int an = answers[i];
            if (an == a[i%5]) sa++;
            if (an == b[i%8]) sb++;
            if (an == c[i%10]) sc++;
        }
        int max = Math.max(sa, Math.max(sb, sc));
        ArrayList<Integer> answer = new ArrayList<>();
        if (max == sa) answer.add(1);
        if (max == sb) answer.add(2);
        if (max == sc) answer.add(3);
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}